### DISCUSSION TITLE
zh/guides/parking-management/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#11849] [Docs]
# zh/guides/parking-management/

使用停车场管理系统Ultralytics YOLOv8

https://docs.ultralytics.com/zh/guides/parking-management/

**Comment by lilinebo** [comment]
I have redrawn the frame and used yolov8n.pt, but the effect is not ideal and there are many recognition errors. It seems that I need to retrain the model?

**Reply by lilinebo** [reply]
![1715321727364](https://github.com/ultralytics/ultralytics/assets/139750690/bccf8b6f-8e8a-48e6-a9b4-5b37f015bff8)

**Reply by lilinebo** [reply]
If I want to train with my own dataset, how should I annotate it, just annotate "car"? Or do the vacant parking spaces also need to be labeled as "nocar"?

**Reply by glenn-jocher** [reply]
For training a model like YOLOv8 on a parking management task, annotations are crucial for teaching the model what to recognize. You should label not just "car" for occupied spaces but also consider labeling "nocar" for vacant parking spaces. This will allow the model to learn the distinction between occupied and empty spots, improving its effectiveness in real-world applications. Here's a small example of what your annotated data might look like:

- **Car:** For any occupied parking spot.
- **NoCar:** For all visible empty parking spots.

By annotating both, you provide clear examples for the model to learn from, which can significantly enhance its accuracy and reliability in deployment.

**Reply by lilinebo** [reply]
thanks

**Reply by lilinebo** [reply]
I have retrained the model and the results have indeed improved a lot, but running the video detection task on 4090 is quite sluggish. What is the reason?

**Reply by glenn-jocher** [reply]
@mozeqiu123 it sounds great that retraining the model improved your results! Regarding the sluggish performance when running video detection on a 4090 — this could be due to several reasons. To boost speed, consider the following:

1. **Resolution**: If your input video resolution is very high, try reducing it to help decrease processing time per frame.
2. **Batch Size**: Adjust the batch size if your predictions aren't in real-time. Sometimes, smaller batch sizes can be more efficient for real-time detections.
3. **Optimizations**: Ensure that you're using any available GPU optimizations like TensorRT or CUDA to accelerate inference on your 4090.
4. **Model Complexity**: If you are using a very deep or complex model version, consider switching to a lighter model like YOLOv8n-lite, which may offer faster inference speeds with good accuracy.

Optimizing these areas can often resolve slowdowns and improve the frame processing rate significantly. If you've already considered these and still face issues, could you provide more details about your setup and the exact command or code you're using? This will help in pinpointing the problem more precisely!
